extension/content/customPersonaEditor.js

```diff
@@ -213,12 +213,25 @@ var CustomPersonaEditor = {
     let property = control.id;
 
     let fp = Cc["@mozilla.org/filepicker;1"].createInstance(Ci.nsIFilePicker);
+    fp.appendFilter("Images", "*.png; *.jpg");
     fp.init(window,
             this._strings.get("backgroundPickerDialogTitle"),
             Ci.nsIFilePicker.modeOpen);
     let result = fp.show();
 
     if (result == Ci.nsIFilePicker.returnOK) {
+      var img = new Image();
+      img.src = fp.fileURL.spec;
+      img.onload = function() {
+          if (img.naturalWidth != 3000 && img.naturalHeight != 200) {
```

Comment:
I think you want `||` instead of `&&` here.

Also, please use non-casting comparators, ie `!==`.